cpp/include/cudf/ast/detail/operators.hpp

```diff
@@ -819,7 +820,17 @@ struct operator_functor<ast_operator::NOT, false> {
 template <typename To>
 struct cast {
   static constexpr auto arity{1};
-  template <typename From>
+  template <typename From, typename cuda::std::enable_if_t<is_fixed_point<From>()>* = nullptr>
+  __device__ inline auto operator()(From f) -> To
+  {
+    if constexpr (cuda::std::is_floating_point_v<To>) {
+      return convert_fixed_to_floating<To>(f);
+    } else {
+      return static_cast<To>(f);
+    }
+  }
+
+  template <typename From, typename cuda::std::enable_if_t<!is_fixed_point<From>()>* = nullptr>
   __device__ inline auto operator()(From f) -> decltype(static_cast<To>(f))
   {
```

Comment:
*question*: Do we need sfinae if the dispatch is `constexpr`? Is the following simpler without having any negative consequences on code size/specialisation?
```
template<typename From>
__device__ inline auto operator()(From f) -> To 
{
  if constexpr (cuda::std::is_floating_point_v<To>) {
    return convert_fixed_to_floating<To>(f);
  } else {
    return static_cast<To>(f);
  }
}
```

Reply:
We need the 2 separate cases because the code is relying on that decltype(static_cast<To>(f)) to SFINAE-out would-be casts like string_view -> double that would otherwise cause compile-time errors.  And of course the static_cast doesn't work for fixed_point anymore.

Reply:
Ah thanks, missed that part.